trio/tests/test_exports.py

```diff
@@ -18,3 +25,25 @@ def test_core_is_properly_reexported():
                 found += 1
         print(symbol, found)
         assert found == 1
+
+
+def test_pylint_sees_all_non_underscore_symbols_in_namespace():
+    # Test pylints ast to contain the same content as dir(trio)
+    linter = PyLinter()
+    ast_set = set(linter.get_ast(trio.__file__, 'trio'))
+    trio_set = set([symbol for symbol in dir(trio) if symbol[0] != '_'])
+    trio_set.remove('tests')
+    assert trio_set - ast_set == set([])
+
+
+@pytest.mark.skipif(
+    sys.version_info.minor > 7 and sys.version_info.releaselevel != 'final',
+    reason="jedi test for python 3.8-dev not supported"
+)
```

Comment:
On further thought... instead of using `skipif` here, I think this is a case where it's better to follow the ["EAFP principle"](https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python). So we should always run the test, but wrap the call to `jedi.Script` in a `try`, like:

```python
try:
    script = jedi.Script("import trio; trio.")
except NotImplementedError:  # pragma: no cover
    pytest.skip("jedi does not yet support this version of python")
```

The advantage of doing it this way, is that it automatically works for all versions of Python – as soon as jedi starts supporting 3.8, the test will start running, and when we start testing on 3.9-dev, we won't have to adjust anything...